Modules/Challenges/src/Challenges.API.Client/API/ChallengesClient.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Net;
+using Challenges.API.Client.Models;
+using RestSharp;
+
+namespace Challenges.API.Client.API;
+public class ChallengesClient
+{
+    private readonly RestClient _client;
+
+    public ChallengesClient()
+    {
+        _client = new RestClient("http://localhost:5000");
+
+        ServicePointManager.ServerCertificateValidationCallback +=
+                (sender, cert, chain, sslPolicyErrors) => true;
+    }
+
+    public async Task<RestResponse<AccessTokenResponse>> GetAccessTokenAsync(AuthenticationRequest authenticationRequest)
+    {
+        var request = new RestRequest("/connect/token", Method.Post);
+        request.AddParameter("grant_type", "password");
+        request.AddParameter("client_id", authenticationRequest.ClientId);
+        request.AddParameter("client_secret", authenticationRequest.ClientSecret);
+        request.AddParameter("username", authenticationRequest.Username);
+        request.AddParameter("password", authenticationRequest.Password);
+
+        return await _client.ExecuteAsync<AccessTokenResponse>(request);
+    }
+
+    public async Task<RestResponse<ChallengeResponse>> CreateChallengeAsync(string accessToken = default)
```

Comment:
As a user of the Client you shouldn't need to pass an access token. Instead, the client should get one if there is none or if the existing one is expired.
There are probably some http clients out there which have built-in support for OAuth, and which help you managing the access token.